lib/lhm/atomic_switcher.rb

```diff
@@ -44,7 +50,21 @@ def validate
     private
 
     def execute
-      @connection.sql(statements)
+      begin
+        @connection.sql(statements)
+      rescue ActiveRecord::StatementInvalid => error
+        if should_retry_exception?(error) && (@retries += 1) < @max_retries
```

Comment:
Better than before and the alt provided only +1s when it actually needs to retry